test/blackbox-tests/test-cases/pkg/ocamllsp/dev-tool-ocamllsp-relock-on-ocaml-version-change.t

```diff
@@ -0,0 +1,64 @@
+Test that if the version of the "ocaml" package in the project's
+lockdir changes then the ocamllsp dev tool is re-locked to be built
+with the version of the ocaml compiler now in the project's
+lockdir. This is necessary because ocamllsp must be compiled with the
+same version of the ocaml compiler as the code that it's analyzing.
+
+  $ . ../helpers.sh
+  $ . ./helpers.sh
+
+  $ mkrepo
+  $ make_mock_ocamllsp_package
+  $ mkpkg ocaml 5.2.0
+  $ mkpkg ocaml 5.1.0
+
+  $ setup_ocamllsp_workspace
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.16)
+  > 
+  > (package
+  >  (name foo)
+  >  (allow_empty))
+  > EOF
+
+  $ make_lockdir
+  $ cat > dune.lock/ocaml.pkg <<EOF
+  > (version 5.2.0)
+  > EOF
+
+Initially ocamllsp will be depend on ocaml.5.2.0 to match the project.
+  $ DUNE_CONFIG__PKG_BUILD_PROGRESS=enabled dune tools exec ocamllsp
```

Comment:
```suggestion
  $ dune tools exec ocamllsp
```
Do we need to print the progress ?  I think the way we print progress may change.